### PR TITLE
don't fail if setuptools < 18.0 but Cython >= 0.23.4

### PR DESCRIPTION
**Pull request by anthrotype**
sorry Frederik if I keep reworking this, but I wasn't happy about requiring such a recent version of setuptools (>=18.0) in order to run the `setup.py`.

One thing is telling user to install a package (say, Cython), another one is to tell them to upgrade their package manager. I also experimented with bootstrapping setuptools via `ez_setup.py` method, but that also felt a bit intrusive.

The fact is setuptools>=18.0 is only required if one doesn't already have Cython installed when running booleanOperation's setup.

So I made the requirement a bit looser and clarified the situation in README.md. 

So, to sum up: 

- either one first installs the current Cython, and *then* run booleanOperations' setup.py;
- or one ensures that setuptools is up-to-date, and let setuptools fetch Cython requirement.

I hope this is clear now.

### CODE REVIEW

**Comment by anthrotype** [comment]
it's important to note that *no dependency* is needed if one builds booleanOperations from the source distribution (.tar.gz or .zip). This is why I would like to tag a new 0.2 release, so that we can upload a new sdist to the Github Releases page, and people won't need to clone the git repo, require Cython, etc.

I think this should be the preferred approach for regular users (vs developers).

**Comment by typemytype** [comment]
Ive tagged a 0.2 release

everything should now go automatically? adding those downloads and builds for different OS's?

**Comment by anthrotype** [comment]
good! 
I'm afraid, it's not gonna be automatic.
The .yml configuration files are already present, however the automatic builds will not trigger until you enable the repository from your Travis and Appveyor projects page. Unfortunately I can't do it for you.

**Comment by anthrotype** [comment]
You can sign in to Travis and Appveyor using your Github credentials, then you can enable the booleanOperations repo from the list of repositories that are connected with your Github account.

**Comment by typemytype** [comment]
done + made a small change to trigger all connected services

hope this works...

gr Frederik
www.typemytype.com

**Comment by anthrotype** [comment]
nice!

https://ci.appveyor.com/project/typemytype/booleanoperations
https://travis-ci.org/typemytype/booleanOperations

Now, the CI will trigger on any push or pull request.
For the automatic deployment of the compiled packages to Github Releases we also need to add the secure Github API key to both .yml files. It's commented out now.

You need to generate a new secret token from your https://github.com/settings/tokens -- don't share it with anybody! --, encrypt it (I'll explain how below) and then paste it in the .yml files.

For Travis, they recommend installing the `travis` command line tool (`sudo gem install travis`) and, from the root of the repo, run the command `travis setup releases`

https://docs.travis-ci.com/user/deployment/releases/

For Appveyor, you can encrypt your token from the website, by going to **Account -> Encrypt** page:

http://www.appveyor.com/docs/deployment/github

Let me know if I can help.

**Comment by typemytype** [comment]
I cannot install travis

I have a newer Ruby and it fails with long traceback...

gr Frederik
www.typemytype.com

**Comment by anthrotype** [comment]
hmm.. 
installing the gem seems to be the only way to encrypt the variables: 
https://docs.travis-ci.com/user/encryption-keys/

**Comment by anthrotype** [comment]
with Node.js there's also a `travis-encrypt` package:

```
npm install pwmckenna/node-travis-encrypt
```

https://www.npmjs.com/package/travis-encrypt

you can download Node from here https://nodejs.org/en/

**Comment by anthrotype** [comment]
I managed to install `travis-encrypt` command line tool with the above npm command.

it installs it to a `node_modules` in the current working directory. You'll find the script in `node_modules/travis-encrypt/bin/travis-encrypt-cli.js`. I made symlink to my /usr/local/bin so I can call it simply with `travis-encrypt`. 

You use it by doing this:
```
$ travis-encrypt --repo typemytype/booleanOperations GITHUB_API_KEY=abcdefghijklmnopqrstuvxyz
```

Take the output and add it to your .travis.yml file: 

```
deploy:
  provider: releases
  api_key:
    secure: "XqJtWxYjtjhRO3[...]"
```

**Comment by typemytype** [comment]
got it working and pushed an update, hope this works...

**Comment by anthrotype** [comment]
excellent! :100: 
travis already pushed the wheels. Now let's wait for Appveyor (that usually takes a bit longer)

**Comment by anthrotype** [comment]
yeah, it's working! Good stuff :dancer:
